### PR TITLE
chore: move more components

### PR DESCRIPTION
**Pull request by layershifter**
# BREAKING CHANGES

Only specific props are now passed to component's style functions.

| `Checkbox`      |
| --------------- |
| `checked`       |
| `disabled`      |
| `labelPosition` |
| `toggle`        |

| `Icon`     |
| ---------- |
| `bordered` |
| `circular` |
| `color`    |
| `disabled` |
| `name`     |
| `outline`  |
| `rotate`   |
| `size`     |
| `xSpacing` |
| `isFontIcon` |
| `isSvgIcon` |

| `Label`             |
| ------------------- |
| `hasActionableIcon` |
| `hasImage`          |
| `circular`          |
| `color` |
| `imagePosition`     |

| `Slider`   |
| ---------- |
| `fluid`    |
| `vertical` |
| `disabled` |

| `Status` |
| -------- |
| `color`  |
| `size`   |
| `state`  |

| `Text`      |
| ----------- |
| `atMention` |
| `color`     |
| `important` |
| `timestamp` |
| `truncated` |
| `disabled`  |
| `error`     |
| `success`   |
| `temporary` |
| `align`     |
| `weight`    |
| `size`      |

### CODE REVIEW

File: packages/react/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -63,116 +78,146 @@ export interface CheckboxProps extends UIComponentProps, ChildrenComponentProps
...
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(Checkbox.handledProps, props)
 
-  handleChange = (e: React.ChangeEvent) => {
+  const handleChange = (e: React.ChangeEvent) => {
     // Checkbox component doesn't present any `input` component in markup, however all of our
     // components should handle events transparently.
-    const { disabled } = this.props
-    const checked = !this.state.checked
+    const checked = !state.checked
```
Comment (mnajdova) [inline packages/react/src/components/Checkbox/Checkbox.tsx]: _nit:_ move this line in the condition below

File: packages/react/src/components/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -63,116 +78,146 @@ export interface CheckboxProps extends UIComponentProps, ChildrenComponentProps
...
+      actions.toggle(checked)
+      _.invoke(props, 'onChange', e, { ...props, checked })
     }
   }
 
-  handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
-    const { disabled } = this.props
-    const checked = !this.state.checked
+  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
+    const checked = !state.checked
```
Comment (mnajdova) [inline packages/react/src/components/Checkbox/Checkbox.tsx]: nit: move this line in the condition below

File: packages/react/src/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -63,116 +78,146 @@ export interface CheckboxProps extends UIComponentProps, ChildrenComponentProps
...
+      _.invoke(props, 'onClick', e, { ...props, checked })
+      _.invoke(props, 'onChange', e, { ...props, checked })
     }
   }
 
-  handleFocus = (e: React.FocusEvent) => {
-    _.invoke(this.props, 'onFocus', e, this.props)
-  }
+  const labelElement = Text.create(label, {
+    defaultProps: () => ({
```
Comment (mnajdova) [inline packages/react/src/components/Checkbox/Checkbox.tsx]: No `getA11Props`?

File: packages/react/src/components/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -63,116 +78,146 @@ export interface CheckboxProps extends UIComponentProps, ChildrenComponentProps
...
+    <ElementType
+      {...getA11Props('root', {
+        className: classes.root,
+        onClick: handleClick,
+        onChange: handleChange,
+        ...unhandledProps,
+      })}
+    >
+      {labelPosition === 'start' && labelElement}
+      {Icon.create(icon, {
```
Comment (mnajdova) [inline packages/react/src/components/Checkbox/Checkbox.tsx:180]: No `getA11Props`?

File: packages/react/src/components/Icon/Icon.tsx [hunk 5]
```diff
@@ -45,50 +67,97 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
   xSpacing?: IconXSpacing
 }
 
-class Icon extends UIComponent<WithAsProp<IconProps>, any> {
-  static create: ShorthandFactory<IconProps>
+const Icon: React.FC<WithAsProp<IconProps>> & FluentComponentStaticProps = props => {
+  const {
```
Comment (mnajdova) [inline packages/react/src/components/Icon/Icon.tsx:76]: Let's destruct the props after the `setStart()`

File: packages/react/src/components/Icon/Icon.tsx [hunk 6]
```diff
@@ -45,50 +67,97 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
...
-      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
-        {isSvgIcon && callable(maybeIcon.icon)({ classes, rtl, props: this.props })}
-      </ElementType>
-    )
-  }
+  setEnd()
+
+  return (
+    <ElementType {...getA11Props('root', { className: classes.root, ...unhandledProps })}>
+      {isSvgIcon && callable(maybeIcon.icon)({ classes, rtl: context.rtl, props })}
```
Comment (mnajdova) [inline packages/react/src/components/Icon/Icon.tsx:131]: Maybe we should restrict the props sent here too. Not as part of this PR, though

File: packages/react/src/components/Icon/Icon.tsx [hunk 7]
```diff
@@ -45,50 +67,97 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
...
-    const maybeIcon = icons[name]
-    const isSvgIcon = maybeIcon && maybeIcon.isSvg
-
-    return (
-      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
-        {isSvgIcon && callable(maybeIcon.icon)({ classes, rtl, props: this.props })}
-      </ElementType>
-    )
-  }
+  setEnd()
```
Comment (mnajdova) [inline packages/react/src/components/Icon/Icon.tsx]: Add the result in an element const and invoke `sentEnd()` right before returning.

File: packages/react/src/components/Label/Label.tsx [hunk 8]
```diff
@@ -52,100 +66,137 @@ export interface LabelProps
...
+  } = props
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: Label.displayName,
+    rtl: context.rtl,
+  })
+  const { classes, styles: resolvedStyles } = useStyles(Label.displayName, {
+    className: Label.className,
+    mapPropsToStyles: () => ({
+      hasActionableIcon: _.has(icon, 'onClick'),
```
Comment (mnajdova) [inline packages/react/src/components/Label/Label.tsx:97]: 👍

File: packages/react/src/components/Label/Label.tsx [hunk 9]
```diff
@@ -52,100 +66,137 @@ export interface LabelProps
...
-            hasEndElement && (
-              <>
-                {endIcon}
-                {endImage}
-              </>
-            )
-          }
-          gap={pxToRem(3)}
-        />
+      <ElementType
```
Comment (mnajdova) [inline packages/react/src/components/Label/Label.tsx:118]: No `setEnd()` invoked

File: packages/react/src/components/Status/Status.tsx [hunk 10]
```diff
@@ -31,46 +38,66 @@ export interface StatusProps extends UIComponentProps {
   state?: 'success' | 'info' | 'warning' | 'error' | 'unknown'
 }
 
-class Status extends UIComponent<WithAsProp<StatusProps>, any> {
-  static create: ShorthandFactory<StatusProps>
-
-  static className = 'ui-status'
+const Status: React.FC<WithAsProp<StatusProps>> & FluentComponentStaticProps = props => {
+  const { className, color, icon, size, state, design, styles, variables } = props
```
Comment (mnajdova) [inline packages/react/src/components/Status/Status.tsx]: Add telemetry

File: packages/react/test/specs/commonTests/handlesAccessibility.tsx [hunk 11]
```diff
@@ -133,8 +134,10 @@ export default (
         wrapper.setProps({})
       }
 
-      getEventTargetComponent(component, 'onKeyDown').simulate('keydown', {
-        keyCode: keyboardKey.Enter,
+      act(() => {
```
Comment (mnajdova) [inline packages/react/test/specs/commonTests/handlesAccessibility.tsx:137]: Is this change necessary to be part of this PR? We can improve tests in a separate PR

File: packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap [hunk 12]
```diff
@@ -124,7 +124,7 @@ exports[`felaRenderer marginLeft is rendered into marginLeft due to RTL with \`n
 
 
 <div className=ui-provider__box dir=rtl>
-  <span className=ui-text a dir=auto>
+  <span dir=auto className=ui-text a>
```
Comment (mnajdova) [inline packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap]: I wonder why these are changed :\

File: packages/react/src/components/Icon/Icon.tsx [hunk 13]
```diff
@@ -45,50 +67,97 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
...
+  })
 
-  static className = 'ui-icon'
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(Icon.handledProps, props)
 
-  static displayName = 'Icon'
+  const { icons = {} } = context.theme
+  const maybeIcon = icons[name]
+  const isSvgIcon = maybeIcon && maybeIcon.isSvg
```
Comment (mnajdova) [inline packages/react/src/components/Icon/Icon.tsx]: This can be sent as a prop in the mapPropsToStyles and simplify the logic in the styles. BTW, I don't see any updates in the iconStyles, please check those too.

File: packages/react/src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (layershifter) [reply packages/react/src/components/Checkbox/Checkbox.tsx]: Makes a sense, done 👍

File: packages/react/src/components/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (layershifter) [reply packages/react/src/components/Checkbox/Checkbox.tsx]: Done 👍

File: packages/react/src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (layershifter) [reply packages/react/src/components/Checkbox/Checkbox.tsx]: Good catch, done 👍

File: packages/react/src/components/Checkbox/Checkbox.tsx (hunk 4 above)
Comment (layershifter) [reply packages/react/src/components/Checkbox/Checkbox.tsx:180]: Good catch, done 👍

File: packages/react/src/components/Icon/Icon.tsx (hunk 5 above)
Comment (layershifter) [reply packages/react/src/components/Icon/Icon.tsx:76]: Done 👍

File: packages/react/src/components/Icon/Icon.tsx (hunk 6 above)
Comment (layershifter) [reply packages/react/src/components/Icon/Icon.tsx:131]: Agree, let's do it separately. I am even not sure that should pass there something

File: packages/react/src/components/Icon/Icon.tsx (hunk 7 above)
Comment (layershifter) [reply packages/react/src/components/Icon/Icon.tsx]: Done 👍

File: packages/react/src/components/Label/Label.tsx (hunk 9 above)
Comment (layershifter) [reply packages/react/src/components/Label/Label.tsx:118]: Good catch, fixed 👍

File: packages/react/src/components/Status/Status.tsx (hunk 10 above)
Comment (layershifter) [reply packages/react/src/components/Status/Status.tsx]: Added 👍

File: packages/react/test/specs/commonTests/handlesAccessibility.tsx (hunk 11 above)
Comment (layershifter) [reply packages/react/test/specs/commonTests/handlesAccessibility.tsx:137]: Yep, `Checkbox` uses `useStateManager()` and it performs actions with hooks. Otherwise it produces a React error:

```
  console.error node_modules/react-dom/cjs/react-dom.development.js:506
    Warning: An update to Checkbox inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
```

File: packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap (hunk 12 above)
Comment (layershifter) [reply packages/react/test/specs/utils/__snapshots__/felaRenderer-test.tsx.snap]: Small issue, I fixed order of props in `Text` and reverted this change 👍

File: packages/react/src/components/Icon/Icon.tsx (hunk 13 above)
Comment (layershifter) [reply packages/react/src/components/Icon/Icon.tsx]: Good catch, did updates there

**Review by miroslavstastny — COMMENTED** [review]
It scares me that `Checkbox` does not render but no UTs fail 😟

File: packages/accessibility/src/behaviors/Icon/iconBehavior.ts [hunk 14]
```diff
@@ -20,8 +20,7 @@ const iconBehavior: Accessibility<IconBehaviorProps> = props => ({
 
 export default iconBehavior
 
-type IconBehaviorProps = {
+export type IconBehaviorProps = {
   /** Alternative text. */
   alt?: string
-  'aria-label'?: string
```
Comment (mnajdova) [inline packages/accessibility/src/behaviors/Icon/iconBehavior.ts:26]: 👍

File: packages/react/src/components/Status/Status.tsx [hunk 15]
```diff
@@ -31,46 +39,70 @@ export interface StatusProps extends UIComponentProps {
...
+    rtl: context.rtl,
+  })
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(Status.handledProps, props)
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      children: false,
-      content: false,
+  const iconElement = Icon.create(icon, {
```
Comment (mnajdova) [inline packages/react/src/components/Status/Status.tsx:70]: No `getA11Props`?

File: packages/react/src/components/Status/Status.tsx [hunk 16]
```diff
@@ -31,46 +39,70 @@ export interface StatusProps extends UIComponentProps {
...
+  const { setStart, setEnd } = useTelemetry(Icon.displayName, context.telemetry)
+  setStart()
 
-  static className = 'ui-status'
-
-  static displayName = 'Status'
+  const { className, color, icon, size, state, design, styles, variables } = props
+  const { classes, styles: resolvedStyles } = useStyles(Status.displayName, {
+    className: Status.className,
+    mapPropsToStyles: () => ({
```
Comment (mnajdova) [inline packages/react/src/components/Status/Status.tsx:50]: Let's update the `statusStyles` with the new typings

File: packages/react/src/themes/teams/types.ts [hunk 17]
```diff
@@ -106,15 +106,15 @@ export type TeamsThemeStylesProps = {
   Reaction?: ReactionProps
   ReactionGroup?: ReactionGroupProps
   Segment?: SegmentProps
-  Slider?: SliderProps
+  Slider?: SliderStylesProps
```
Comment (mnajdova) [inline packages/react/src/themes/teams/types.ts:109]: Let's update `Status` too.

**Review by mnajdova — APPROVED** [review]
Please update https://github.com/microsoft/fluent-ui-react/issues/2269 after merging
